sdk/sdk-skeleton.go

```diff
@@ -0,0 +1,126 @@
+import "time"
+
+type Asset struct{}
+
+//// Methods directly on the SDK
+
+func NewChannel() {
+
+}
+
+func NewChannelListenerService() {
+
+}
+
+func GetChannel() {
+
+}
+
+//// Channel State Machine
+
+type ChannelStateMachine struct {
+	ChannelID              string
+	Status                 string // waiting | active | closing | closed
+	Asset                  Asset
+	InitiatorEscrowAccount string
+	ResponderClaimAmount   int
+	ReserveEscrowAccount   string
+	InitiatorClaimAmount   int
+	IsInitiator            bool
+	MyAccount              string
+	MyClaimAMount          int
+	OtherAccount           string
+	OtherClaimAmount       int
+}
+
+func (c *ChannelStateMachine) Init() {
+	return nil
+}
+
+func (c *ChannelStateMachine) RegisterMonitoringNotificationHandler(handler MonitoringNotificationHandler) error {
+	return nil
+}
```

Comment:
I think we should leave this out while we're focused on the state machine. Any handlers sound like orchestration.

Reply:
https://github.com/stellar/experimental-payment-channels/pull/25/commits/a7322ba7b51ff9ecf88e807adc9401641821a8a3